pages/balance-budget.vue

```diff
@@ -267,31 +183,79 @@ import FiscalYearSelect from "@/components/FiscalYearSelect";
 import Header from "@/components/Header";
 import Footer from "@/components/Footer";
 import DepartmentsWalkthrough from "@/components/DepartmentsWalkthrough";
+import { D3PieChart } from 'vue-d3-charts';
 
+var total_expenses = 1234.0
+var starting_expense = total_expenses / 7.0
+var current_sum_expenses = 1234.0
+ 
 export default Vue.extend({
   components: {
     CitySelect,
     FiscalYearSelect,
     Header,
     Footer,
-    DepartmentsWalkthrough
+    DepartmentsWalkthrough,
+    D3PieChart
+  },
+  mounted() {
+      this.is_mounted = true;
+      this.exceeds_limit = false;
   },
   data() {
-    return {
-      selected_city: "San Francisco",
-      selected_year: "2020-2021",
-      cities: ["San Francisco", "Oakland"],
-      years: [
-        "2015-2016",
-        "2016-2017",
-        "2017-2018",
-        "2018-2019",
-        "2019-2020",
-        "2020-2021"
-      ]
-    };
+        return {
+            total_expenses: total_expenses,
+            selected_city: "San Francisco",
+            selected_year: "2020-2021",
+            cities: ["San Francisco", "Oakland"],
+            years: ["2015-2016","2016-2017","2017-2018","2018-2019","2019-2020","2020-2021"],
+            is_mounted: false,
+            budget_pie_chart_data: [
+                {name: "Community Health", total: starting_expense, dept_color: '#2A6465'},
+                {name: "Culture & Recreation", total: starting_expense, dept_color: '#EF896E'},
+                {name: "General Admin & Finance", total: starting_expense, dept_color: '#F5BD41'},
+                {name: "General City Responsibilities", total: starting_expense, dept_color: '#CAAA97'},
+                {name: "Human Welfare & Neighborhood Development", total: starting_expense, dept_color: '#4DA54A'},
+                {name: "Public Protection", total: starting_expense, dept_color: '#4296AD'},
+                {name: "Public Works, Transportation & Commerce", total: starting_expense, dept_color: '#CF722A'}
+            ],
+            budget_pie_chart_config: {
+                key: "name",
+                value: "total",
+                color: {key: "dept_color"},
+                margin: {left: 100, right: 100},
+                transition: {duration: 100, ease: "easeLinear"}
+            },
+            slider_max: total_expenses,
+            slider_min: 0,
+            health_value: starting_expense,
+            culture_value: starting_expense,
+            admin_value: starting_expense,
+            city_value: starting_expense,
+            welfare_value: starting_expense,
+            protection_value: starting_expense,
+            transport_value: starting_expense,
+            test_rule: [v => current_sum_expenses <= total_expenses]
```

Comment:
nit: naming could be better. what does this do? (not rhetorical, truly wondering what it does :laughing: )

Reply:
haha agreed naming could be better here, will adjust -- this rule is passed through to the v-slider component for validating when the user inputted sum of expenses exceeds revenue